content/docs/extending-tina/format-and-parse.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Format and parse input
+id: '/docs/extending-tina/format-and-parse'
+prev: /docs/extending-tina/customize-list-ui
+next: /docs/extending-tina/customizing-forms
+---
+
+Format and parse can be used together to customize the value of the form (What the users see's) and the value of the data being saved. See the following example,
+
+```ts
+// .tina/schema.{js,ts,tsx}
+// ...fields
+{
+ name: "username",
+ type: "string",
+ label: "Username",
+ ui: {
+// is called on every form change but result is stored in data and not in the form value (saved to file but not displayed to the user)
+   parse: (val?: string)=>val && val.toUpperCase(),
+// Is called on every form change and the result is put back into the value of the form (displayed to the user)           
+   format: (val?: string)=> (val ? val.toLowerCase() : ""),
+ },
+},
+```
+
+As you type you will see all lowercase but when it is saved it will be all uppercase.
```

Comment:
```suggestion
For this example, as the user types, they will see all lowercase characters. When it is saved, it will be all uppercase characters.
```